Makefile

```diff
@@ -256,6 +256,18 @@ image-all: $(addprefix sub-image-,$(VALIDARCHES))
 sub-image-%:
 	$(MAKE) image ARCH=$*
 
+## Test building images for all supported ARCHes
+test-image-all: $(addprefix test-sub-image-,$(VALIDARCHES))
```

Comment:
Why do we need a "test-image-all", is there something wrong with "image-all"?

Reply:
@caseydavenport had mentioned that storage might fill up on semaphore from building all images, so "test-image-all" and "test-sub-image-<arch>" are there just to clean up after building each image.

Reply:
That's understandable, but what we really want to test is image-all, if that target changes we need to now update this target as well.

I'd say test the image-all target, since that's the goal, and if you need to clean up then just clean up the images, as opposed to redefining image-all and it's sub images.

Reply:
Thanks @Brian-McM! I changed the 'sub-image-%' target to conditionally clean based on an env var and am using it on the CI job.